src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs

```diff
@@ -40,49 +40,55 @@ public VirtualPathData GetVirtualPath([NotNull] VirtualPathContext context)
             MvcServicesHelper.ThrowIfMvcNotRegistered(services);
 
             EnsureLogger(context.HttpContext);
-            using (_logger.BeginScope("MvcRouteHandler.RouteAsync"))
-            {
-                var actionSelector = services.GetRequiredService<IActionSelector>();
-                var actionDescriptor = await actionSelector.SelectAsync(context);
+            var actionSelector = services.GetRequiredService<IActionSelector>();
+            var actionDescriptor = await actionSelector.SelectAsync(context);
 
-                if (actionDescriptor == null)
+            if (actionDescriptor == null)
+            {
+                string currentRouteValues = string.Empty;
+                if (context.RouteData != null
+                    && context.RouteData.Values != null)
                 {
-                    LogActionSelection(actionSelected: false, actionInvoked: false, handled: context.IsHandled);
-                    return;
+                    currentRouteValues = string.Join(", ",
+                        context.RouteData.Values.Select(rv => $"Key={rv.Key};Value={rv.Value}"));
```

Comment:
Let's not do this. We don't want to pack values into strings. Let's leave it out for now.